src/bzz-link.ts

```diff
@@ -23,20 +27,100 @@ export function requestFilter(pathname: string, req: Request): boolean {
   return req.subdomains.length >= 1
 }
 
+export function getDomain(req: Request): string | undefined {
+  try {
+    const matches = req.headers.host!.match(/^([^?#]*)(\?([^#]*))?(#(.*))?:[0-9]*]?/i) // clean host domain string
```

Comment:
Why do you do it like this? IMHO you should use [`req.hostname`](https://expressjs.com/en/api.html#req.hostname) as I have already mentioned previously several times.